### ISSUE TITLE
SpotMenu crashes Spotify on macOS 10.14 Mojave Beta

### ISSUE THREAD

**Issue by JacobKingDev** [kmikiy/SpotMenu#126]
### Prerequisites

- macOS 10.14 Mojave Beta
- Spotify
- SpotMenu

### Description

Having SpotMenu and Spotify launched at the same time causes Spotify to crash. In the event that SpotMenu is open first, Spotify simply won't launch and crashes every time you try and open it. In the case that Spotify is open first, opening SpotMenu crashes Spotify.

### Steps to Reproduce

1. Install both programs on macOS Mojave.
2. Open one of them (either is fine, see above).
3. Open the other.

**Expected behavior:** No crash, SpotMenu works as expected.

**Actual behavior:** Spotify crashes.

**Reproduces how often:** 100% of the time.

### Versions

SpotMenu version: 1.8.0
MacOS version: 10.14 Mojave Developer Beta 1

### Additional Information
*None*

**Comment by kmikiy** [comment]
Could you try running these apple scripts and check if they crash Spotify?

```scpt
tell application "Spotify" to get artist of current track
```
```scpt
tell application "Spotify" to get name of current track
```

This can be done from the Script Editor.app. Copy&Paste the snippet and press play.

![screen shot 2018-06-12 at 10 53 28 am](https://user-images.githubusercontent.com/14206503/41280414-28129e68-6e2f-11e8-86ad-c1a357c9b48a.png)

**Comment by harryyoud** [comment]
These run fine for me here (on Mojave 10.14 Mojave Beta, Spotify version 1.0.82.447.g975ad224 and SpotMenu version 1.8.0), but SpotMenu still crashes. Screenshots attached
![screenshot 2018-06-17 at 18 53 20](https://user-images.githubusercontent.com/10576381/41510628-fe735ff4-725f-11e8-9c02-3857f941d683.png)
![screenshot 2018-06-17 at 18 55 10](https://user-images.githubusercontent.com/10576381/41510629-fe88dfc8-725f-11e8-974a-d870a0670890.png)
Please let me know if you want anymore details/debugging/logs

**Comment by kmikiy** [comment]
do you guys have the same issue? SpotMenu crashes Spotify, right?

**Comment by harryyoud** [comment]
Apologies, I meant to say Spotify crashed. We have exactly the same symptoms. Running Spotify in a terminal shows the executable crashes with `Segmentation fault: 11`. I also have a `dtruss` trace of the Spotify binary if you'd like it

**Comment by harryyoud** [comment]
After quite a few repeated launches, I managed to get Spotify to launch fairly inconsistently (perhaps 1 in 5 runs or so)

I have a Spotify logs for both a failure and success case if you'd like them, but I'd much prefer to give you these privately given the possibility of tokens or other personal info present

**Comment by kmikiy** [comment]
yeah, if you could send it that would be nice. my email is m.kristyan@gmail.com 
i'll also try to install mojave on a virtual machine, but i think my computer won't be able to handle it.

**Comment by kmikiy** [comment]
I have set up a virtual machine (works pretty poorly though). I checked other similar apps (e.g. https://github.com/paulyoung/Statusfy) and they also cause Spotify to crash. 😕 I think for some reason the Apple Script API calls cause Spotify to crash, iTunes Apple Script API calls work fine though. I still haven't figured out what the problem is.

**Comment by JacobKingDev** [comment]
Thanks for your quick investigation @kmikiy. If it's not a glaringly obvious issue it may be worth waiting until the GM for Mojave is made available, as weird quirks have a habit of mysteriously disappearing in later betas... Not only that, but if iTunes works fine, it could be an issue for the Spotify devs (not that they'll be particularly eager to fix issues for 3rd party integrations). Otherwise, it could be an entitlement based permission hence Apple app working fine but Spotify not.

**Comment by JacobKingDev** [comment]
Or potentially, a user issue. Who does SpotMenu execute the AppleScript as? Could be that it needs to come from the current user.

**Comment by Nayuta-Kani** [comment]
Hmmm, for me Spotify does not crash though, might be due to the fact I enabled Spotify for "Application Data" and "Accessibility" manually by adding it to the list in Settings > Security & Privacy

**Comment by harryyoud** [comment]
Can confirm this seems to be fixed on Developer Preview 2. I was also asked if I would "Allow SpotMenu to communicate with Spotify" on the first boot after the update, which I haven't seen before. After that, Spotify doesn't seem to crash anymore

**Comment by abustamam** [comment]
Just confirming that I just installed SpotMenu for the first time on Mojave and it works without any hitches. I did see the "Allow SpotMenu to communicate with Spotify" message, but works without a hitch.

**Comment by LosoTheCreator** [comment]
I can also confirm not hitches, glitches, or issues. works perfect on my machine. 
[2017 iMac | Mojave ]

> Just confirming that I just installed SpotMenu for the first time on Mojave (official release) and it works without any hitches. I did see the "Allow SpotMenu to communicate with Spotify" message, but works without a hitch.

**Comment by Hum4n01d** [comment]
Close issue?

**Comment by jacas** [comment]
Just installed SpotMenu and works fine on Mojave. Thank you!

**Comment by srhrshr** [comment]
Can confirm that it works like a charm on Mojave. Thanks :)
```🐶 [9:20] ➜ ~ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.14
BuildVersion:	18A391

**Comment by D-T-Design** [comment]
Working for me on Mojave 10.14.1

**Comment by joshuanathanson** [comment]
Works great on Mojave 10.14.4

**Comment by paulbaker3** [comment]
Working great on Mojave 10.14.5

**Comment by kmikiy** [comment]
closing because I'm rewriting the app from scratch https://github.com/kmikiy/SpotMenu/issues/279#issuecomment-2987565341
